### DISCUSSION TITLE
Signal renames: Consistent between press, select, & squeeze

### DISCUSSION THREAD

**Discussion by comods** [godotengine/godot-proposals#4095] [Scripting]
The press API has not aged well as similar APIs have been added with inconsistent names.

My reasoning for consistency (weighed against each other): Reduced confusion (what timing? differentiated verb), consistent implied details (started what? cognitive load), cleaner reusable code (can: n in variant group: n.enabled = true), ease of knowing more features (A == B, A or B needed?), reduced double negatives, and reduced typing.

The press category (includes select & squeeze) is currently an Outlier from API naming. It is subjective if their category difference is justified based on how someone prioritizes the above competing reasons.

This Proposal is for the category to be consistent within itself, for the above reasons.
Note: WebXRInterface is already not a letter for letter match of JS API.

**Comment by comods** [comment]
**Rename AA:**

AA is for this category to no longer be an Outlier, but consistent with itself and the API as a whole.

BaseButton:
**press_started** (from button_down)
**pressed** (exists as a sibling)
**press_ended** (from button_up) # Docs: always called?

WebXRInterface:
**select_started** (from selectstart)
**selected** (from select) # Docs: not called if aborted
**select_ended** (from selectend) # Docs: always called?
**squeeze_started** (from squeezestart)
 **squeezed** from (squeeze) # Docs: not called if aborted
**squeeze_ended** (from squeezeend) ->  # Docs: always called?

TouchScreenButton:
**pressed** (exists as a sibling)
**press_ended** (from released)

XRController3D:
**button_pressed** (exists as a sibling)
**button_press_ended** (from button_released)

XRPositionalTracker:
**button_pressed** (exists as a sibling)
**button_press_ended** (from button_released)

**Comment by comods** [comment]
Rename BB:

BB is for using released and invoked.

BaseButton:
**pressed** (from button_down)
**invoked** (from pressed)
**released** (from button_up) # Docs: always called?
**invoke_mode** (from action_mode) # on press or release

WebXRInterface:
**select_pressed** (from selectstart)
**select_invoked** (from select) # Docs: not called if aborted
**select_released** (from selectend) # Docs: always called?
**squeeze_pressed** (from squeezestart)
**squeeze_invoked** from (squeeze) # Docs: not called if aborted
**squeeze_released** (from squeezeend) -> # Docs: always called?

**Comment by dsnopek** [comment]
> Note: WebXRInterface is already not a letter for letter match of JS API.

For these particular signal names, it is a letter-for-letter match! See:

https://github.com/godotengine/godot/blob/3.x/modules/webxr/native/library_godot_webxr.js#L282

My goal was for the API to stick as closely to the JS API as we could for everything that didn't need to conform to Godot's XR APIs, so that we could simply reference the JS API for any questions about how it worked.

If there are places where we aren't matching the JS API, I'd actually more prefer to go the other way, and change `WebXRInterface` to more closely conform to the JS.
